### PR TITLE
Subset the list of all alleles to class I and class II before processing further

### LINKED ISSUE #194: Narrow down list of alleles to class I and class II before executing MHCI or MHCII pipelines

### PR DESCRIPTION
**Pull request by susannasiebert**
Closes #194

### CODE REVIEW

File: pvacseq/lib/prediction_class.py [hunk 1]
```diff
@@ -76,6 +75,13 @@ class MHCI(PredictionClass, metaclass=ABCMeta):
     def url(self):
         return 'http://tools-api.iedb.org/tools_api/mhci/'
 
+    @classmethod
+    def prediction_classes(cls):
+        prediction_classes = []
+        for prediction_class in cls.__subclasses__():
+            prediction_classes.append(prediction_class)
+        return prediction_classes
```
Comment (tmooney) [inline pvacseq/lib/prediction_class.py]: Is this loop and return different from `return cls.__subclasses__()`?

Comment (susannasiebert) [inline pvacseq/lib/prediction_class.py]: It's not. I pushed a commit to simplify this.

**Comment by chrisamiller** [comment]
What happens if the allele you enter is invalid and is neither class I or class II? Is that already taken care of in code outside the scope of this PR?

**Comment by susannasiebert** [comment]
`PredictionClass.check_alleles_valid(args.allele)` on line 185 will check for that.

**Comment by chrisamiller** [comment]
Nice. +1

**Pull request merged**